sphinx/writers/html.py

```diff
@@ -41,4 +41,11 @@ def translate(self) -> None:
                      'footer', 'html_prolog', 'html_head', 'html_title',
                      'html_subtitle', 'html_body'):
             setattr(self, attr, getattr(visitor, attr, None))
-        self.clean_meta = ''.join(self.visitor.meta[2:])
+        self.clean_meta = ''.join([
+            meta_tag for meta_tag in self.visitor.meta
+            if not any(skip_tag in meta_tag for skip_tag in (
+                'charset="utf-8"',
+                'name="generator"',
+                'name="viewport"',
```

Comment:
The tag values are of type `str`, so for now I'm using string-contains comparisons here.  An alternative could be to parse them as HTML markup and then check the names/attributes..

Reply:
Is it possible for the quoted values to be quoted using single quotes? Also, maybe you could check before the join().

Reply:
That would be valid, yep.  I'm usually reluctant to use regular expressions, and I felt that using `charset`, `viewport` and `generator` instead could risk some false positives.. and that parsing the HTML is kinda heavyweight for what's required here, so I went with a simple approach intending for exactly this kind of discussion.